plonky2x/core/src/backend/function/mod.rs

```diff
@@ -171,6 +175,8 @@ impl<C: Circuit> Plonky2xFunction for C {
         );
 
         if let PublicOutput::Bytes(output_bytes) = output {
+            // TODO: can optimize this by saving the wrapped circuit in build and loading it from a binary
```

Comment:
Good observation--in practice, this may not impact performance at all as it doesn't take too long to build the wrapper circuits JIT.